modules/dashboard/php/dashboard.class.inc

```diff
@@ -268,34 +268,36 @@ class Dashboard extends \NDB_Form
                  ORDER BY Testdate DESC LIMIT 4",
                 array()
             );
-            for ($i=0; $i < sizeof($bvl_feedback); $i++) {
-                if ($bvl_feedback[$i]['Testdate'] > $last_login) {
-                    $bvl_feedback[$i]['new'] = 1;
+            $frontend_feedback = array();
+            foreach ($bvl_feedback as $row) {
+                if ($row['Testdate'] > $last_login) {
+                    $row['new'] = 1;
                 } else {
-                    $bvl_feedback[$i]['new'] = 0;
+                    $row['new'] = 0;
                 }
-                if ($bvl_feedback[$i]['Feedback_level'] === 'profile') {
-                    $bvl_feedback[$i]['URL'] = '/' . $bvl_feedback[$i]['CandID'];
-                } else if ($bvl_feedback[$i]['Feedback_level'] === 'visit') {
-                    $bvl_feedback[$i]['URL'] = '/instrument_list/?candID='
-                        . $bvl_feedback[$i]['CandID'] . '&sessionID='
-                        . $bvl_feedback[$i]['SessionID'];
-                } else if ($bvl_feedback[$i]['Feedback_level'] === 'instrument') {
+                if ($row['Feedback_level'] === 'profile') {
+                    $row['URL'] = '/' . $row['CandID'];
+                } else if ($row['Feedback_level'] === 'visit') {
+                    $row['URL'] = '/instrument_list/?candID='
+                        . $row['CandID'] . '&sessionID='
+                        . $row['SessionID'];
+                } else if ($row['Feedback_level'] === 'instrument') {
```

Comment:
These all look like they're changing behaviour significantly.. why is `$bvl_feedback[$i]` changed to `$row`?

Reply:
I think Phan gave some sort of error about getting an `int` as an array key when it was expecting another type. I can't remember exactly.

I think the behaviour should be the same though.